### PR TITLE
Final project submissions

### PR DESCRIPTION
**Pull request by aradradev**
This pull request includes the full setup and implementation of project requirements for the React app.

### Config & Basic Setup
- [x] Create React App.
- [x] Install React Redux, Redux Logger, and React Router.
- [x] Download the free image for the app logo.
- [x] Create routes and view components: rockets, missions, my profile.
- [x] Use `<NavLink />` for the page navigation links and style active class to indicate the current section/page.

### Redux: Fetch Data and Update Redux Store
- [x] Fetch data from the SpaceX API endpoints:
  - Rockets: https://api.spacexdata.com/v4/rockets
  - Missions: https://api.spacexdata.com/v3/missions
- [x] Store the selected data in the Redux store:
  - Rockets:
    - id
    - rocket_name
    - description
    - flickr_images
  - Missions:
    - mission_id
    - mission_name
    - description

### Final Submission
- [x] Basic setup and configuration completed.
- [x] Routes and view components created for Rockets, Missions, and My Profile pages.
- [x] Navigation links implemented using `<NavLink />`.
- [x] Active class styled to indicate the currently active page.
- [x] Redux state slices created for rockets, missions, and dragons (if applicable).
- [x] Data fetched from SpaceX API endpoints for rockets and missions.
- [x] Fetched data stored in the Redux store.
- [x] Rockets data stored in Redux includes id, rocket_name, description, and flickr_images.
- [x] Missions data stored in Redux includes mission_id, mission_name, and description.

## Summary
The project has been successfully implemented with all the required features. It utilizes Create React App for the basic setup and React Redux for state management. The Redux store stores data fetched from the SpaceX API for rockets and missions, ensuring efficient data handling. Navigation links and active class styling provide a smooth user experience. The app is now ready for review.

Please review the PR and provide your valuable feedback. Thank you!

### CODE REVIEW

File: src/tests/Mission.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+it('matches snapshot', () => {
+  const tree = renderer
+    .create(
+      <Provider store={store}>
+        <MissionsList />
+      </Provider>,
+    )
+    .toJSON();
+  expect(tree).toMatchSnapshot();
```
Comment (elisha2kyakpo1) [inline src/tests/Mission.test.js:15]: - [ ] You did a great job adding these snapshots, but as per project instructions, kindly test all the components and functions. You can test to see if certain elements such as a title, button, or text are on the page(browser).

test

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @aradradev, @Rifat8080, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Tests passing(snapshots)👏
👏 The app works as expected.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Mission.test.js (hunk 1 above)
Comment (aradradev) [reply src/tests/Mission.test.js:15]: @elisha2kyakpo1 Can you clear that for me, we tested all the components as a group of two persons we only have three tests the Rockets.js, Missions.js and MyProfile.js

**Comment by tugsem** [comment]
## STATUS: APPROVED ✔️ 
Hi @aradradev and @Rifat8080 , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Alt Text
Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**